src/jit/gentree.h

```diff
@@ -2341,6 +2313,173 @@ struct GenTreePhi final : public GenTree
+{
+    class Use
+    {
+        GenTree*  m_node;
+        Use*      m_next;
+        unsigned  m_offset;
+        var_types m_type;
+
+    public:
+        Use(GenTree* node, unsigned offset, var_types type)
+            : m_node(node), m_next(nullptr), m_offset(offset), m_type(type)
+        {
+        }
+
+        GenTree*& NodeRef()
+        {
+            return m_node;
+        }
+
+        GenTree* GetNode() const
+        {
+            return m_node;
+        }
+
+        void SetNode(GenTree* node)
+        {
+            assert(node != nullptr);
+            m_node = node;
+        }
+
+        Use*& NextRef()
+        {
+            return m_next;
+        }
+
+        Use* GetNext() const
+        {
+            return m_next;
+        }
+
+        void SetNext(Use* next)
+        {
+            m_next = next;
+        }
+
+        unsigned GetOffset() const
+        {
+            return m_offset;
+        }
+
+        var_types GetType() const
+        {
+            return m_type;
+        }
+
+        void SetType(var_types type)
+        {
+            m_type = type;
+        }
+    };
+
+    class UseIterator
+    {
+        Use* use;
+
+    public:
+        UseIterator(Use* use) : use(use)
+        {
+        }
+
+        Use& operator*()
+        {
+            return *use;
+        }
+
+        Use* operator->()
+        {
+            return use;
+        }
+
+        void operator++()
+        {
+            use = use->GetNext();
+        }
+
+        bool operator==(const UseIterator& other)
+        {
+            return use == other.use;
+        }
+
+        bool operator!=(const UseIterator& other)
+        {
+            return use != other.use;
+        }
+    };
+
+    class UseList
+    {
+        Use* head;
+
+    public:
+        UseList(Use* head) : head(head)
+        {
+        }
+
+        UseIterator begin() const
+        {
+            return head;
+        }
+
+        UseIterator end() const
+        {
+            return nullptr;
+        }
+    };
+
+    Use* gtUses;
+
+    // TODO-Cleanup: It would make more sense for GT_FIELD_LIST's type to be TYP_STRUCT.
+    // TYP_INT or the type of the first field is used instead to match the old implementation.
+    GenTreeFieldList() : GenTree(GT_FIELD_LIST, TYP_INT), gtUses(nullptr)
+    {
+        SetContained();
+    }
+
+    UseList Uses()
+    {
+        return gtUses;
+    }
+
+    //--------------------------------------------------------------------------
+    // Equals: Check if 2 FIELD_LIST nodes are equal.
+    //
+    // Arguments:
+    //    list1 - The first FIELD_LIST node
+    //    list2 - The second FIELD_LIST node
+    //
+    // Return Value:
+    //    true if the 2 FIELD_LIST nodes have the same type, number of uses, and the
+    //    uses are equal.
+    //
+    static bool Equals(GenTreeFieldList* list1, GenTreeFieldList* list2)
+    {
+        // TODO-Cleanup: The list type is actually ignored. It should always be TYP_STRUCT
```

Comment:
Currently this is only called from `Compare` and that does its own flag check already. That check is kind of weird: https://github.com/dotnet/coreclr/blob/f7776685fb9ee16f7aa11b74a08abd35eb671277/src/jit/gentree.cpp#L1248-L1252

Even if we want to be able to use `Equals` directly (IMO it would be a reasonable thing to do) it's not clear what flags should be checked.